portal-impl/src/com/liferay/portal/service/impl/LayoutLocalServiceImpl.java

```diff
@@ -1209,9 +1209,50 @@ public Layout getLayoutByIconImageId(long iconImageId)
 	public Map<Long, List<Layout>> getLayoutChildLayouts(
 		LayoutSet layoutSet, List<Layout> parentLayouts) {
 
-		List<Layout> childLayouts = _getChildLayouts(
-			layoutSet,
-			ListUtil.toLongArray(parentLayouts, Layout::getLayoutId));
+		Map<LayoutSet, List<Layout>> parentLayoutMap = new HashMap<>();
```

Comment:
When virtual layouts are involved, not all layouts listed in List<Layout> parentLayouts come from the LayoutSet specified in the parameter (they can come from a different source LayoutSet). This code sorts them into the proper buckets.